04.class/operateInterface.js

```diff
@@ -0,0 +1,88 @@
+import readline from "readline";
+import enquirer from "enquirer";
+
+const { Select } = enquirer;
+
+export default class OperateInterface {
+  constructor(memoOperation) {
+    this.memoOperation = memoOperation;
+  }
+
+  async listMemos() {
+    const memos = await this.memoOperation.fetchMemos();
+    if (this.#isMemoEmpty(memos)) return;
+    memos.forEach((memo) => {
+      console.log(memo.content.split("\n")[0]);
+    });
+  }
+
+  async readMemos() {
+    const content = await this.#selectMemoFromList("see");
+    if (!content) return;
+    console.log(content);
+  }
+
+  async deleteMemo() {
+    const id = await this.#selectMemoFromList("delete");
+    if (!id) return;
+    this.memoOperation.removeMemo(id);
+    console.log("Memo deleted!");
+  }
+
+  async writeMemo() {
+    const content = await this.#imputMemo();
+    await this.memoOperation.addMemo(content);
+    console.log("Memo added.");
+  }
+
+  async #selectMemoFromList(action) {
+    const memos = await this.memoOperation.fetchMemos();
+    if (this.#isMemoEmpty(memos)) return;
+    const choices = this.#extractFirstLineAndValue(memos, action);
+    return this.#promptForMemo(choices, action);
+  }
+
+  #isMemoEmpty(memos) {
+    if (memos.length === 0) {
+      console.error("No memos found.");
+      return true;
+    }
+  }
+
+  #extractFirstLineAndValue(memos, action) {
+    return memos.map((memo) => {
+      const firstLine = memo.content.split("\n")[0] || " ";
```

Comment:
https://github.com/hirokiej/js-practices/pull/5#discussion_r1903917294 の返信を読みました。
なるほどー、と思いつつ、そもそも1行目が空のメモというのはふつう作らないと思うので、いったん何も考慮しない（コードではなくデータが悪いと考える）でもいいのかなーと思いました（コードをなるべくシンプルに保つ）。

```suggestion
      const firstLine = memo.content.split("\n")[0];
```

Reply:
上記で修正します！
確かに、空で作る人は普通はいないです…！
空で作られそうになった場合は、エラーを出力する処理をしたほうがアプリとして自然でした。